internal/rpcserver/rpcserver.go

```diff
@@ -4026,6 +4057,58 @@ func handlePing(_ context.Context, s *Server, cmd interface{}) (interface{}, err
 	return nil, nil
 }
 
+// handleReconsiderBlock implements the reconsiderblock command.
+func handleReconsiderBlock(_ context.Context, s *Server, cmd interface{}) (interface{}, error) {
+	c := cmd.(*types.ReconsiderBlockCmd)
+	hash, err := chainhash.NewHashFromStr(c.BlockHash)
+	if err != nil {
+		return nil, rpcDecodeHexError(c.BlockHash)
+	}
+
+	chain := s.cfg.Chain
+	err = chain.ReconsiderBlock(hash)
+	if err != nil {
+		if errors.Is(err, blockchain.ErrUnknownBlock) {
+			return nil, &dcrjson.RPCError{
+				Code:    dcrjson.ErrRPCBlockNotFound,
+				Message: fmt.Sprintf("Block not found: %v", hash),
+			}
+		}
+
+		// Use separate error code for failed validation.
+		allRuleErrs := func(err error) bool {
+			var rErr blockchain.RuleError
+			if !errors.As(err, &rErr) {
+				return false
+			}
+
+			var mErr blockchain.MultiError
+			if errors.As(err, &mErr) {
+				for _, e := range mErr {
+					if !errors.As(e, &rErr) {
+						return false
+					}
+				}
+			}
+
+			return true
+		}
```

Comment:
I went back and forth on this.  I originally had added an `AsAll` to `blockchain.MultiError`, but the caller still had to check if it's a `MultiError` first to then invoke that method, so it didn't really change anything about this code in the end. 

On the other hand, I considered a function that took an error, but then the behavior was pretty ambiguous to callers.  e.g. What if there is a wrapped error that is not of the target type but itself also wraps one that is?  Maybe that is what the caller wants and maybe it isn't.  It's pretty specific to what the function that returned the error con produce.

Given all of that, and since this is the only place that it's really needed, I just opted to do it inline.

I'm open to suggestions here though.

Reply:
Makes sense.  Sounds good to leave it as is then, and we can reconsider moving it if we end up needing similar logic in other places.